examples/jsm/tsl/display/FXAANode.js

```diff
@@ -34,279 +34,257 @@ class FXAANode extends TempNode {
-				relativeVContrast.mulAssign( fxaaQualityinvEdgeThreshold );
+			const pLuminance = select( isHorizontal, ln, le );
+			const nLuminance = select( isHorizontal, ls, lw );
+			const pGradient = abs( pLuminance.sub( lm ) );
+			const nGradient = abs( nLuminance.sub( lm ) );
 
-				If( abs( relativeVContrast ).lessThan( 0.3 ), () => { // 45 deg edge
+			const pixelStep = select( isHorizontal, texSize.y.negate(), texSize.x ).toVar();
+			const oppositeLuminance = float().toVar();
+			const gradient = float().toVar();
 
-					// 1 1 .
-					// 0 0 1
-					// . 0 1
+			If( pGradient.lessThan( nGradient ), () => {
+				
+				pixelStep.assign( pixelStep.negate() );
+				oppositeLuminance.assign( nLuminance );
+				gradient.assign( nGradient );
 
-					// do a simple blur
-					const sum = rgbaN.add( rgbaS ).add( rgbaE ).add( rgbaW );
-					return mix( rgbaM, sum.mul( 0.25 ), 0.4 );
+			} ).Else( () => {
 
-				} );
+				oppositeLuminance.assign( pLuminance );
+				gradient.assign( pGradient );
 
 			} );
 
-			const offNP = vec2().toVar();
+			return vec4( isHorizontal, pixelStep, oppositeLuminance, gradient );
+
+		} );
 
-			If( relativeVContrast.lessThanEqual( 0 ), () => {
+		const DetermineEdgeBlendFactor = Fn( ( [ texSize, lm, isHorizontal, pixelStep, oppositeLuminance, gradient, uv ] ) => {
 
-				rgbaN.assign( rgbaW );
-				rgbaS.assign( rgbaE );
+			const uvEdge = uv.toVar();
+			const edgeStep = vec2().toVar();
+			If( isHorizontal, () => {
 
-				// . 0 .      1
-				// 1 0 1  ->  0
-				// . 0 .      1
+				uvEdge.addAssign( vec2(0, pixelStep.mul( 0.5 ) ) );
+				edgeStep.assign( vec2( texSize.x, 0.0 ) );
 
-				offNP.x.assign( 0 );
-				offNP.y.assign( fxaaQualityRcpFrame.y );
+			} ).Else( () => {
 
-			 } ).Else( () => {
+				uvEdge.addAssign( vec2( pixelStep.mul( 0.5 ), 0) );
+				edgeStep.assign( vec2 ( 0.0, texSize.y ) );
 
-				offNP.x.assign( fxaaQualityRcpFrame.x );
-				offNP.y.assign( 0 );
+			} );
 
-			 } );
+			const edgeLuminance = lm.add( oppositeLuminance ).mul( 0.5 );
+			const gradientThreshold = gradient.mul( 0.25 );
 
-			const mn = contrast( rgbaM, rgbaN ).toVar();
-			const ms = contrast( rgbaM, rgbaS ).toVar();
+			const puv = uvEdge.add( edgeStep.mul( EDGE_STEPS.element( 0 ) ) ).toVar();
+			const pLuminanceDelta = SampleLuminance( puv ).sub( edgeLuminance ).toVar();
+			const pAtEnd = abs( pLuminanceDelta ).greaterThanEqual( gradientThreshold ).toVar();
 
-			If( mn.lessThanEqual( ms ), () => {
+			Loop( EDGE_STEP_COUNT, ( { i } ) => {
 
-				rgbaN.assign( rgbaS );
+				If( pAtEnd, () => { Break(); } );
 
+				puv.addAssign( edgeStep.mul( EDGE_STEPS.element( i ) ) );
+				pLuminanceDelta.assign( SampleLuminance( puv ).sub( edgeLuminance ) );
+				pAtEnd.assign( abs( pLuminanceDelta ).greaterThanEqual( gradientThreshold ) );
+				
 			} );
 
-			const doneN = int( 0 ).toVar();
-			const doneP = int( 0 ).toVar();
 
-			const nDist = float( 0 ).toVar();
-			const pDist = float( 0 ).toVar();
+			If( !pAtEnd, () => {
```

Comment:
`!pAtEnd` it's not a currently TSL syntax, try `pAtEnd.not()`

Reply:
That did it! I got 100% match 🙏🙏